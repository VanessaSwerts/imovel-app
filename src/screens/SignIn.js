import React from 'react'
import { KeyboardAvoidingView, StyleSheet, Text, TouchableOpacity, Image } from 'react-native'

import logo from '../../assets/img/house_agreement.png'
import colors from '../constants/colors.json'

import InputArea from '../components/InputArea'

export default function SignIn({ navigation }) {
  return (
    <KeyboardAvoidingView
      style={styles.container}
      behavior="padding"
      enabled={Platform.OS === 'ios'}
    >
      <Image style={styles.logo} source={logo}></Image>
      <Text style={styles.title} >IMovel</Text>

      <InputArea icon={'envelope'} placeholder={'Entre com o seu email'} keyboardType={'email-address'} />
      <InputArea icon={'lock'} placeholder={'Entre com a sua senha'} password={true} />

      <TouchableOpacity style={styles.button} onPress={() => navigation.navigate('Home')}>
        <Text style={styles.buttonText}>Entrar</Text>
      </TouchableOpacity>
    </KeyboardAvoidingView>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors["platinum"],
    justifyContent: 'center',
    alignItems: 'center',
    padding: 35
  },

  logo: {
    width: 110,
    height: 110,
  },

  title: {
    fontSize: 48,
    fontWeight: "500",
    color: colors['blue-secondary'],
    textAlign: 'center',
  },

  button: {
    height: 45,
    backgroundColor: colors['yellow'],
    borderRadius: 24,
    marginTop: 10,
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'stretch',
  },

  buttonText: {
    color: colors['blue'],
    fontWeight: 'bold',
    fontSize: 16
  }
})