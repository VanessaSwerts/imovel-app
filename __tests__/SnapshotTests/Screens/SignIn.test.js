import React from 'react'
import { act, create } from 'react-test-renderer'
import SignIn from '../../../src/screens/SignIn'

jest.useFakeTimers()
jest.mock('react-native-vector-icons/FontAwesome', () => 'Icon')

describe('<SignIn />', () => {
  it('render correctly', () => {
    act(() => { tree = create(<SignIn />) })

    expect(tree.toJSON()).toMatchSnapshot()
  })

})