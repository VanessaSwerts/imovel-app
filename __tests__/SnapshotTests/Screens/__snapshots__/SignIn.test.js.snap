// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> render correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#E5E5E5",
        "flex": 1,
        "justifyContent": "center",
        "padding": 35,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 110,
        "width": 110,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#19294d",
        "fontSize": 48,
        "fontWeight": "500",
        "textAlign": "center",
      }
    }
  >
    IMovel
  </Text>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFF",
        "borderColor": "#14213D",
        "borderRadius": 24,
        "borderWidth": 1,
        "elevation": 2,
        "flexDirection": "row",
        "height": 46,
        "marginTop": 15,
        "paddingHorizontal": 12,
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2,
      }
    }
  >
    <Icon
      color="#14213D"
      name="envelope"
      size={16}
      style={
        Object {
          "alignSelf": "center",
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      keyboardType="email-address"
      placeholder="Entre com o seu email"
      placeholderTextColor="#d2d2d2"
      rejectResponderTermination={true}
      style={
        Object {
          "paddingHorizontal": 8,
          "paddingVertical": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFF",
        "borderColor": "#14213D",
        "borderRadius": 24,
        "borderWidth": 1,
        "elevation": 2,
        "flexDirection": "row",
        "height": 46,
        "marginTop": 15,
        "paddingHorizontal": 12,
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2,
      }
    }
  >
    <Icon
      color="#14213D"
      name="lock"
      size={16}
      style={
        Object {
          "alignSelf": "center",
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      placeholder="Entre com a sua senha"
      placeholderTextColor="#d2d2d2"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "paddingHorizontal": 8,
          "paddingVertical": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#FCA311",
        "borderRadius": 24,
        "height": 45,
        "justifyContent": "center",
        "marginTop": 10,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#14213D",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Entrar
    </Text>
  </View>
</View>
`;
